### ISSUE TITLE
Quota of dockerhub is still used in v2.1.1 after the image is cached.

### ISSUE THREAD

**Issue by reasonerjt** [goharbor/harbor#13566]
Was reported by Remoco Breekveldt from slack channel:

1. Harbor Version v2.1.1-5f52168e with Proxy cache configured on example.com/docker-hub
2. Enter the Harbor Core pod and check the remaing rate limit with the following commands:
```
export HEADER='Authorization: Bearer'
export TOKEN=$(curl "https://auth.docker.io/token?service=registry.docker.io&scope=repository:ratelimitpreview/test:pull" | sed 's/{"token":"//g' | sed 's/","access_token":.*//g') 
curl --head -H "$HEADER $TOKEN" https://registry-1.docker.io/v2/ratelimitpreview/test/manifests/latest 2>&1
```
Remaining-limit is showing 250
3. From a local machine loop a pull for busybox using the following command (note: ratelimit is 250 today):
for i in {1..255}; do docker pull example.com/docker-hub/library/busybox; done
4. Enter the Harbor Core pod and check the remaing rate limit with the following commands:
```
export HEADER='Authorization: Bearer'
export TOKEN=$(curl "https://auth.docker.io/token?service=registry.docker.io&scope=repository:ratelimitpreview/test:pull" | sed 's/{"token":"//g' | sed 's/","access_token":.*//g') 
curl --head -H "$HEADER $TOKEN" https://registry-1.docker.io/v2/ratelimitpreview/test/manifests/latest 2>&1
```
5. We noticed that we consumed all our rate-limit and actually the remaining limit is 'minus' 5

**Comment by perryrh0dan** [comment]
I have the same problem.

**Comment by reasonerjt** [comment]
Seems the root cause is that the client in the adapter accepts v1 manifests:
https://github.com/goharbor/harbor/blob/c3fde4e483a1fdcc7a4cd4c0407007406d622790/src/pkg/registry/client.go#L51

And dockerhub will serve v1 manifests, so the digest in the header of response to HEAD request will always mismatch the digest in Harbor which is the digest of v2 manifests.

Making the accept header configurable and force docker hub to return v2 manifests may likely fix the problem.

**Comment by xaleeks** [comment]
unfortunately everything on dockerhub is a manifest list.  Caching everything is not a robust solution as storage would blow up way too fast.  What we need to do is compare the digests of the matching OS image only; whether ARM redis has been updated is irrelevant to me if I'm on centos.  

Thanks to the users who brought this to our attention.  investigating a fix now

**Comment by reasonerjt** [comment]
We'll try to cache the complete manifest-list in another place (probably redis) and compare the digest and the complete manifest list in `UseLocal` method, meanwhile we need to make sure the current flow to write the manifest list (index.json) remain unchanged, because the index.json is used in other artifacts like CNAB.

**Comment by xaleeks** [comment]
I want to highlight making this work for docker images without having to store unnecessary images is higher priority than supporting CNAB.

**Comment by xaleeks** [comment]
this will be fixed in the v2.2 release, without having to cache non-matching OS images
